src/lxc/lxccontainer.c

```diff
@@ -4698,6 +4719,299 @@ static bool do_lxcapi_restore(struct lxc_container *c, char *directory, bool ver
 
 WRAP_API_2(bool, lxcapi_restore, char *, bool)
 
+static int ensure_dummy_dir(const char *dest)
+{
+	int ret;
+	struct stat sb;
+
+	ret = stat(dest, &sb);
+	if (ret == 0) {
+		if (S_ISDIR(sb.st_mode))
+			return 0;
+
+		ret = unlink(dest);
+		if (ret < 0) {
+			SYSERROR("Failed to remove \"%s\"", dest);
+			return -1;
+		}
+	}
+
+	ret = mkdir(dest, 0000);
+	if (ret < 0) {
+		SYSERROR("Failed to create directory \"%s\"", dest);
+		return -1;
+	}
+
+	return 0;
+}
+
+static int ensure_dummy_file(const char *dest)
+{
+	int ret;
+	struct stat sb;
+
+	ret = stat(dest, &sb);
+	if (ret == 0) {
+		if (!S_ISDIR(sb.st_mode))
+			return 0;
+
+		ret = rmdir(dest);
+		if (ret < 0) {
+			SYSERROR("Failed to remove \"%s\"", dest);
+			return -1;
+		}
+	}
+
+	ret = mknod(dest, S_IFREG | 0000, 0);
+	if (ret < 0) {
+		SYSERROR("Failed to create file \"%s\"", dest);
+		return -1;
+	}
+
+	return 0;
+}
+
+/* @st_mode is the st_mode field of the stat(source) return struct */
+static int create_mount_target(mode_t st_mode, const char *dest)
+{
+	char *dirdup, *destdirname;
+	int ret;
+
+	dirdup = strdup(dest);
+	if (!dirdup) {
+		SYSERROR("Failed to duplicate target name \"%s\"", dest);
+		return -1;
+	}
+	destdirname = dirname(dirdup);
+
+	ret = mkdir_p(destdirname, 0755);
+	if (ret < 0) {
+		SYSERROR("Failed to create \"%s\"", destdirname);
+		free(dirdup);
+		return ret;
+	}
+	free(dirdup);
+
+	if (S_ISDIR(st_mode))
+		ret = ensure_dummy_dir(dest);
+	else
+		ret = ensure_dummy_file(dest);
+	if (ret < 0) {
+		SYSERROR("Failed to create mount target \"%s\"", dest);
+		return -1;
+	}
+
+	return 0;
+}
+
+static int do_lxcapi_mount(struct lxc_container *c, const char *source,
+			   const char *target, const char *filesystemtype,
+			   unsigned long mountflags, const void *data,
+			   struct lxc_mount *mnt)
+{
+	char *suff, *sret;
+	size_t len;
+	char template[MAXPATHLEN], path[MAXPATHLEN];
+	pid_t pid, init_pid;
+	struct stat sb;
+	int ret = -1, fd = -EBADF;
+
+	if (!c || !c->lxc_conf) {
+		ERROR("Container or configuration is NULL");
+		return -EINVAL;
+	}
+
+	if (!c->lxc_conf->lxc_shmount.path_host) {
+		ERROR("Host path to shared mountpoint must be specified in the config\n");
+		return -EINVAL;
+	}
+	len = strlen(c->lxc_conf->lxc_shmount.path_host) + sizeof("/.lxcmount_XXXXXX") - 1;
+
+	ret = snprintf(template, len + 1, "%s/.lxcmount_XXXXXX", c->lxc_conf->lxc_shmount.path_host);
```

Comment:
This `snprintf` + `strlen` pattern is the same as the one I previously commented on, the strlen part could be simplified away I think.